### ISSUE TITLE
[fix] Focus Trap for Modal Component

### ISSUE THREAD

**Issue by DanielMiller-agile6** [trussworks/react-uswds#212]
I've been experimenting with this library on a personal project and really enjoying it so far!  I ran into an issue with the Modal component and didn't see an existing issue for it, so figured I'd log it and see if this has been considered already.

**Does your feature request relate to a specific USWDS feature?**
No

**Is your feature request related to a problem? Please describe.**
In the "Full Example" story for the Modal component, user focus isn't trapped inside of the modal dialog.  I'm able to tab focus out of the modal to elements behind the overlay.  This doesn't comply with WAI-ARIA Authoring practices for modal dialogs:

https://www.w3.org/TR/wai-aria-practices/#dialog_modal

**Describe the solution you'd like**
I'm wondering if it makes sense to include a "focus trap" utility inside of this library, similar to how [reach-ui](https://reacttraining.com/reach-ui/) handles this for their [modal dialogs](https://github.com/reach/reach-ui/blob/master/packages/dialog/src/index.tsx).  This could either be by using an existing 3rd party library, or by implementing a custom one specifically for `react-uswds`.

**Describe alternatives you've considered**
Another possibility might be to leave this behavior up to the consumer, and provide user guidance in documentation along with another example story.  You could also consider leveraging reach-ui as a basis for this component, although that would require a significant additional dependency.

**Additional context**
None

**Comment by suzubara** [comment]
Taking a look at https://github.com/reach/reach-ui/blob/develop/packages/dialog/src/index.tsx and  https://github.com/theKashey/react-focus-lock I think handling our Modal similarly is the way to go here!

**Comment by suzubara** [comment]
After discussing with the team, we decided that the Modal component should be deprecated from this library, since it was designed internally and is not part of USWDS. We will plan to move it into a different package as soon as possible. I created related issues https://github.com/trussworks/react-uswds/issues/786 and https://github.com/trussworks/react-uswds/issues/787 and will close this issue as won't fix.
